openmdao/components/add_subtract_comp.py

```diff
@@ -150,58 +158,67 @@ def add_equation(self, output_name, input_names, vec_size=1, length=1, val=1.0,
         res_ref : float or ndarray
             Scaling parameter. The value in the user-defined res_units of this output's residual
             when the scaled value is 1. Default is 1.
+        append : bool
+            If True, add the given equation to the list.  If False, add the appropriate I/O but
+            do not append the given equation to the internal list.
         """
         kwargs = {'units': units, 'res_units': res_units, 'desc': desc,
                   'lower': lower, 'upper': upper, 'ref': ref, 'ref0': ref0,
                   'res_ref': res_ref}
-        self._add_systems.append((output_name, input_names, vec_size, length, val,
-                                  scaling_factors, kwargs))
 
-    def add_output(self):
-        """
-        Use add_equation instead of add_output to define equation systems.
-        """
-        raise NotImplementedError(self.msginfo + ': Use add_equation method, not add_output '
-                                  'method to create an addition/subtraction relation')
+        if isinstance(input_names, str):
+            input_names = input_names
```

Comment:
did you mean `input_names = [input_names]` here?

Reply:
actually, a length 1 input_names set makes no sense... so I changes it to an error condition